### PR TITLE
TPS-454: Bubble Map Chart

### PR DESCRIPTION
**Pull request by cwbuecheler**
**Description**
This PR adds the long-anticipated clickable/scrollable map chart with markers and marker clusters. It allows users to define where their lat/long data comes from, a metric to display in the tooltip, a radius at which markers cluster (defined as pixels on the map, so adjusts as they zoom), cluster and marker colors, and a few other things.

**Note:** the dev postgresql database has been updated to provide continuous geolocation data, so you should be able to test locally if you're connected to that DB.

**Acceptance Criteria**
- Code looks good!
- Map works as expected when paired with geolocation data
- Markers cluster as the user zooms out and break apart as the user zooms in
- Marker / Cluster colors change as anticipated
- Tooltips display metric title and value

**Note:** We don't currently break out our order values by city, so every city within a given country will display the same count in the tooltip. This is not a bug so much as a data issue that should be solvable by clients but isn't worth adjusting our own sample data for.

**Examples**
https://www.loom.com/share/4e42216271e0498b9e26d43064adfb5b?sid=e1ac580b-6479-465d-ad2c-c7fc73d2d990

### CODE REVIEW

File: src/components/vanilla/charts/BubbleMapChart/index.tsx [hunk 1]
```diff
@@ -0,0 +1,202 @@
+// React & 3rd Party
+import React, { useEffect, useState } from 'react';
+import { DataResponse, Dataset, DimensionOrMeasure, Measure } from '@embeddable.com/core';
+import Leaflet, { LatLngBoundsExpression, MarkerCluster } from 'leaflet';
+import { MapContainerProps, MarkerProps, TileLayerProps, Tooltip, useMap } from 'react-leaflet';
```
Comment (mad-raccoon) [inline src/components/vanilla/charts/BubbleMapChart/index.tsx:5]: "react-leaflet" doesn't include the same imports as the "leaflet" library?

Comment (cwbuecheler) [reply src/components/vanilla/charts/BubbleMapChart/index.tsx:5]: Unfortunately, no. See here for info: https://react-leaflet.js.org/docs/start-installation/

Comment (cwbuecheler) [reply src/components/vanilla/charts/BubbleMapChart/index.tsx:5]: Oh, hmm, actually maybe - let me see if I can reduce those. You still have to have leaflet in the project but may not need to import from it.

Comment (cwbuecheler) [reply src/components/vanilla/charts/BubbleMapChart/index.tsx:5]: @mad-raccoon - confirmed: React-Leaflet does not export those two items.

File: src/components/vanilla/charts/BubbleMapChart/BubbleMapChart.emb.ts [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+      description: 'Dataset',
+      defaultValue: false,
+      category: 'Chart Data',
+    },
+    {
+      name: 'bubblePlacementLat',
+      type: 'dimension',
+      label: 'Bubble Placement - Latitude',
+      config: {
+        dataset: 'ds',
```
Comment (hjkmarshall) [inline src/components/vanilla/charts/BubbleMapChart/BubbleMapChart.emb.ts:25]: Do we have a latitude type?

I"m guessing not, but if so, we could do something like:

supportedTypes: ['latitude']

**Comment by hjkmarshall** [comment]
It's now working nicely for me :)

File: src/components/vanilla/charts/BubbleMapChart/BubbleMapChart.emb.ts (hunk 2 above)
Comment (cwbuecheler) [reply src/components/vanilla/charts/BubbleMapChart/BubbleMapChart.emb.ts:25]: Unfortunately, Cube doesn't provide a latitude or longitude type

**Comment by cwbuecheler** [comment]
@mad-raccoon - I think all issues are addressed and tests are now passing. Could you approve this? Thanks!

**Pull request merged**